crates/floresta-chain/src/pruned_utreexo/udata.rs

```diff
@@ -139,10 +136,20 @@ pub mod proof_util {
         let proof = Proof::new(targets, hashes);
         let mut hashes = Vec::new();
         let mut leaves_iter = udata.leaves.iter().cloned();
-        let mut tx_iter = transactions.iter();
+
+        let tx_iter = block.txdata.iter();
 
         let mut inputs = HashMap::new();
-        tx_iter.next(); // Skip coinbase
+        let actual_block_height = match chain.get_block_height(&block.block_hash()) {
+            Ok(height) => {
+                if let Some(height) = height {
+                    height
+                } else {
+                    return Err(BlockchainError::BlockNotFound);
+                }
+            }
+            Err(_) => return Err(BlockchainError::BlockNotFound),
+        };
 
```

Comment:
This is horrible, because `get_block_height ` returns an `Option` inside a `Result`(????)

Reply:
You can simplify this to 

```rust
       let actual_block_height = chain.get_block_height(&block.block_hash())?.ok_or(BlockchainError::BlockNotFound)?;
```

Reply:
No, i cant because of the return type of the function... generic on errors. 
Can we change the return type on another PR ?

Reply:
That's annoying.

>  Can we change the return type on another PR ?

Yes, we need to improve the error handling in `ChainState`. I'm not sure how. Did you try introducing the `where` clause suggested by the compiler?